src/TestSuite/Schema/SchemaCleaner.php

```diff
@@ -0,0 +1,148 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @since         4.3.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\TestSuite\Schema;
+
+use Cake\Console\ConsoleIo;
+use Cake\Database\Schema\BaseSchema;
+use Cake\Database\Schema\CollectionInterface;
+use Cake\Datasource\ConnectionInterface;
+use Cake\Datasource\ConnectionManager;
+
+/**
+ * This class will help dropping all tables of a given connection
+ * and truncate the non phinx migrations tables.
+ */
+class SchemaCleaner
+{
+    /**
+     * Drop all tables of the provided connection.
+     *
+     * @param string $connectionName Name of the connection.
+     * @param \Cake\Console\ConsoleIo|null $io Console IO to output the processes.
+     * @return void
+     * @throws \Exception if the dropping failed.
+     */
+    public static function drop(string $connectionName, ?ConsoleIo $io = null)
+    {
+        self::info('Dropping all tables for connection ' . $connectionName, $io);
+
+        $schema = static::getSchema($connectionName);
+        $dialect = static::getDialect($connectionName);
+
+        $stmts = [];
+        foreach ($schema->listTables() as $table) {
+            $table = $schema->describe($table);
+            $stmts = array_merge($stmts, $dialect->dropTableSql($table));
+        }
+
+        static::executeStatements(ConnectionManager::get($connectionName), $stmts);
+    }
+
+    /**
+     * Truncate all tables of the provided connection.
+     *
+     * @param string $connectionName Name of the connection.
+     * @param \Cake\Console\ConsoleIo|null $io Console IO to output the processes.
+     * @return void
+     * @throws \Exception if the truncation failed.
+     */
+    public static function truncate(string $connectionName, ?ConsoleIo $io = null)
+    {
+        static::info('Truncating all tables for connection ' . $connectionName, $io);
+
+        $stmts = [];
+        $schema = static::getSchema($connectionName);
+        $dialect = static::getDialect($connectionName);
+        $tables = $schema->listTables();
+        $tables = self::unsetMigrationTables($tables);
+        foreach ($tables as $table) {
+            $table = $schema->describe($table);
+            $stmts = array_merge($stmts, $dialect->truncateTableSql($table));
```

Comment:
Same as above.

Reply:
`instanceof` or `method_exists()` is how we solve this in other places.

Reply:
Agreed, I'll update accordingly